### PR TITLE
fix: handle unicode cursor offsets in language server completion

### PR DESCRIPTION
**Pull request by 0xWeakSheep**
## Summary

  This fixes a language server panic triggered when completion is requested near a Unicode character.

  The root cause was that LSP `Position.character` values were treated like byte offsets. For multibyte characters, that could produce an offset in
  the middle of a UTF-8 code point, which then caused `detect_completion_context` to panic when slicing the source string.

  ## Changes

  - convert LSP UTF-16 columns to byte offsets in `to_offset_from_position`
  - clamp incomplete multibyte positions to the start of the containing character
  - make `detect_completion_context` resilient to non-boundary offsets by flooring to a valid char boundary before slicing
  - add regression tests for Unicode cursor handling

  ## Why this fixes the issue

  The reported crash happened because completion context detection sliced the source text using an invalid byte index. After this change:

  - cursor positions from the client are translated to valid byte offsets
  - invalid intermediate offsets no longer reach string slicing
  - completion context detection has a defensive boundary-safe fallback

  This prevents the panic reported in #1356 while preserving normal completion behavior.

  ## Tests

  Ran successfully:

  - `cargo fmt --all`
  - `cargo test -p fe-language-server to_offset_from_position -- --nocapture`
  - `cargo test -p fe-language-server detect_completion_context -- --nocapture`
  - `cargo test -p fe-language-server completion_snapshot -- --nocapture`

### CODE REVIEW

File: crates/language-server/src/util.rs [hunk 1]
```diff
@@ -21,13 +21,37 @@ pub fn calculate_line_offsets(text: &str) -> Vec<usize> {
...
 pub fn to_offset_from_position(position: Position, text: &str) -> parser::TextSize {
     let line_offsets: Vec<usize> = calculate_line_offsets(text);
-    let line_offset = line_offsets
+    let line_start = line_offsets
         .get(position.line as usize)
         .copied()
         .unwrap_or_else(|| line_offsets.last().copied().unwrap_or(0));
-    let character_offset = position.character as usize;
+    let line_end = line_offsets
+        .get(position.line as usize + 1)
```
[the remark this answers is not in the document]
Comment (0xWeakSheep) [reply crates/language-server/src/util.rs]: Good catch. I fixed this in a follow-up commit by replacing the `position.line as usize + 1` lookup with a saturating next-line lookup.
I also added a regression test covering `Position::new(u32::MAX, 0)` so malformed LSP positions do not panic on 32-bit targets.
